src/containers/BooksForm.js

```diff
@@ -1,20 +1,82 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
 
-const BooksForm = () => {
-  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.baseState = this.state;
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-  return (
-    <form>
-      <label htmlFor="title">
-        Title
-        <input id="title" type="text" name="title" />
-      </label>
-      <select id="category">
-        {categories.map(e => <option key={e} value={e}>{e}</option>)}
-      </select>
-      <button type="submit">Submit</button>
-    </form>
-  );
-};
+  async handleChange(e, field) {
+    await this.setState({
+      [field]: e.target.value,
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const { createBook } = this.props;
+    createBook(this.state);
+    this.setState(this.baseState);
```

Comment:
**OPTIONAL**

This is nice, you are keeping a reference to the default state so you can update it here.

I don't have strong feelings about this, but you might be adding some mental tax to the person maintaining your code, why not keep this simple and remove the mental tax involved on this by setting the state to the desired object that we want?

That way is going to be obvious for the next one picking up your code what you're doing (yourselves included) .

Keep in mind the [KISS](https://en.wikipedia.org/wiki/KISS_principle) principle.